core/sugartypes.ml

```diff
@@ -1,44 +1,7 @@
-(** The syntax tree created by the parser. *)
-
-type name = string [@@deriving show]
```

Comment:
Why have you moved this to `Operators.ml`? It seems to me that you have just disorganised the sugared syntax module.

Reply:
I believe the problem was that `sugartypes.ml` depends on `types.ml` and `lens_phrase` in `types.ml` depends on the operators (which would be in `sugartypes.ml`). The lens types in `typ` depend on `lens_phrase`. The alternative would be copy pasting the operators, which I wanted to avoid.

Reply:
@slindley, @SimonJF this is the part that Daniel was objecting to regarding the operators.  Please have a look.

Reply:
This seems like a fairly standard DAG-ification refactoring that is necessary for this change.

Reply:
I don't see anything wrong with this. I'm guessing @dhil hadn't realised that the operators were pulled out because they are used in two places in order to avoid a cyclic dependency.